app/src/main/java/com/obsidian/plugins/task_notifier/os/AlertManager.kt

```diff
@@ -14,21 +14,22 @@ import java.util.*
 class AlertManager {
   fun syncNotifications(context: Context, reminders: List<ObsidianReminderBO>) {
     cancelAllNotifications(context)
-    val reqIds = createNotifications(context, reminders)
-    PersistenceManager.addActiveAlerts(context, reqIds)
+    val remindersWithReqIds = createNotifications(context, reminders)
+    PersistenceManager.addActiveReminders(context, remindersWithReqIds)
   }
 
   private fun createNotifications(
     context: Context,
     reminders: List<ObsidianReminderBO>
-  ): List<Int> {
+  ): List<ObsidianReminderBO> {
     val reqIds: ArrayList<Int> = ArrayList()
     reminders.forEach { reminder ->
       val reqId = UUID.randomUUID().hashCode()
```

Comment:
I think its a feature to include the past. I mean, if you had a problem with syncing maybe you miss notification and you just realize it when you open the obsidian. 

But i know if it is in the past, but you have 3 file change event then you will get the notification 3 times. (because its sends instantly).

If we calculate the reqId somehow based on what vault, what file, what line changed we can filter out the already sent past event notifications

Reply:
OK we can reduce this case if you edit the vault only alone. Maybe you would not get notification if the sync is later than the notification:I'll fix it